src/main/java/view/NavigationMenu.java

```diff
@@ -0,0 +1,76 @@
+package view;
+
+import app.AppBuilder;
+import interface_adapter.login.LoginState;
+import interface_adapter.navigation.NavController;
+import interface_adapter.navigation.NavViewModel;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+/**
+ * Navigation menu to the application
+ */
+public class NavigationMenu extends JMenuBar implements ActionListener {
+
+    private final JButton buttonSearch = new JButton("Search");
+    private final JButton buttonRecipeCollection = new JButton("Recipe Collection");
+    private final JButton buttonChangePassword = new JButton("Change Password");
+    private final JButton buttonLogout = new JButton("Logout");
+    private NavController navController;
+
+    public NavigationMenu(NavViewModel navViewModel) {
+        this.add(buttonSearch);
+        this.add(buttonRecipeCollection);
+        this.add(buttonChangePassword);
+        this.add(buttonLogout);
+        this.setVisible(true);
```

Comment:
Maybe consider more organization/division of components through the use of JPanels.